### PR TITLE
chore: Use shiny's Makefile to allow for test commands to change over time

### PR DESCRIPTION
**Pull request by schloerke**
Related: 
* posit-dev/py-shiny#1729
* posit-dev/py-shiny#1731

## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [x] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Re lated issue # NA
- Closes # NA

## Checklist

- [ ] Code follows style guide (ruff)
- [x] Tests added 
- [ ] Documented the changes

## If you have comments or can explain your changes, please do so below.

The default `pytest` command does cover some tests, but others are ran under default settings. By integrating the make commands, `py-shiny` can update the internal script as things change without need for updates to `narwhals`.

### CODE REVIEW

**Review by MarcoGorelli — COMMENTED** [review]
nice, thanks @schloerke !

is there a way to keep the `mypy` command tested, to check we don't regress the type annotations?

**Comment by schloerke** [comment]
Good catch! I lost it in the shuffle yesterday.

The `mypy` check within py-shiny's repo is only there to test that it _can_ run (because mypy sometimes doesn't like commented code)... not to find type errors. We do have type coverage using `pyright` that is tested within `make check` (along with other linting checks).

I've changed py-shiny's `make narwhals-test-integration` command to perform both type checking and test all narwhals related unit tests in https://github.com/posit-dev/py-shiny/pull/1733. (Waiting for CI to auto merge in the next few mins)

File: .github/workflows/downstream_tests.yml [hunk 1]
```diff
@@ -113,9 +113,11 @@ jobs:
       - name: install-basics
         run: uv pip install --upgrade tox virtualenv setuptools --system
       - name: install-shiny-dev
-        run: |
+        env:
+          UV_SYSTEM_PYTHON: 1
+        run: |
```
Comment (schloerke) [inline .github/workflows/downstream_tests.yml:118]:
```suggestion
        run: |
```

**Comment by schloerke** [comment]
Working type check: https://github.com/narwhals-dev/narwhals/actions/runs/11293892161/job/31413093004?pr=1161#step:10:48

File: .github/workflows/downstream_tests.yml [hunk 2]
```diff
@@ -125,10 +127,4 @@ jobs:
       - name: Run pytest
```
Comment (schloerke) [inline .github/workflows/downstream_tests.yml:127]:
```suggestion
      - name: Run `make narwhals-test-integration`
```

**Review by MarcoGorelli — APPROVED** [review]
wow amazing, thanks @schloerke !

**Pull request merged**